docs/extend/extend-cli/cli-developing-a-plugin.md

```diff
@@ -2,218 +2,323 @@
+                rejectUnauthorized: params.arguments.rejectUnauthorized,
+                protocol: params.arguments.protocol || "https"
+            };
+            const sessCfgWithCreds = await ConnectionPropsForSessCfg.   addPropsOrPrompt<ISession>(sessCfg, params.arguments,
+                { doPrompting: true, parms: params });
+            const session = new Session(sessCfgWithCreds);
+
+            const resp = await DataSetDiff.diff(session, params.arguments.oldDataSet, params.arguments.newDataSet);
+            params.response.console.log(resp);
+        }
+    }    ```
+
+
+**Note:** If you are adding multiple commands to your CLI plug-in, consider moving the code that creates a session into a base handler class that can be shared across multiple commands. See the [sample plugin code](https://github.com/zowe/zowe-cli-sample-plugin/blob/master/src/cli/list/ListBaseHandler.ts) for an example of how this can be done.
+
+**To create a unique plug-in:** **[what can we advise here?]**
 
 ## Trying your command
-Be sure to build your plug-in via `npm run build`.
 
-Install your plug-in into Zowe CLI via `zowe plugins install`.
+Before you test your new command, confirm that you are able to connect to the mainframe.
+
+In order for the client-side API to reach the mainframe (to fetch data sets), Zowe CLI needs a z/OSMF profile for access. See [Using profiles](https://docs.zowe.org/stable/user-guide/cli-using-using-profiles/) for information. 
+
+Once the connection between Zowe CLI and z/OSMF is confirmed, build and install the plug-in before running it for the first time.
+
+Follow these steps:
+
+1. Repeat the steps in [Building your plug-in source](../extend-cli/cli-developing-a-plugin#building-your-plug-in-source).
+
+2. Issue the `zowe plugins install` command to install Files Util Plug-in into Zowe CLI.
 
-Issue the following command. Replace the data set names with valid mainframe data set names on your system:
+    A success message displays if installed correctly.
 
-![Pipe Output](../../images/guides/CLI/pipeOutput.png)
+3. Issue the following command (the Files Util Plug-in command) replacing the data set names with valid mainframe data set names on your system:
 
-The raw diff output is displayed as a command response:
+    ```
+    zowe files-util diff data-sets "IBMUSER.REXX(TESTADD)" "IBMUSER.REXX(TESTSUB)"
+    ```
+    The raw diff output displays as a command response:
 
-![Raw Diff Output](../../images/guides/CLI/diffOutput.png)
+    ```
+    $ zowe files-util diff data-sets "IBMUSER.REXX(TESTADD)" "IBMUSER.REXX(TESTSUB)"
+    ===================================================================
+    --- IBMUSER.REXX(TESTADD)        Old
+    +++ IBMUSER.REXX(TESTSUB)        New
+    @@ -1,6 +1,6 @@
+    /* REXX */
+    -say add(3,2)
+    +say sub(3,2)
+    exit
+    -add:
+    +sub:
+    parse arg a,b
+    -return a + b
+    +return a - b
+    ```
+
+**To create a unique plug-in:** **[what can we advise here?]**
```

Comment:
Steps 1 and 2 would be the same, but the command to run in step 3 and expected output would change depending on what command you implemented above.